python_modules/libraries/dagster-components/dagster_components/core/dsl_schema.py

```diff
@@ -42,6 +42,13 @@ class AssetAttributesModel(RenderedModel):
         Optional[str], RenderingMetadata(output_type=Optional[AutomationCondition])
     ] = None
 
+    def _render_property(self, key, raw_value, value_resolver):
+        rendered = super()._render_property(key, raw_value, value_resolver)
+        if key == "key":
+            # coerce the string asset key into an AssetKey object
+            return AssetKey.from_user_string(rendered) if rendered else None
+        return rendered
+
```

Comment:
this seems super gross and instead there should more structured and well-typed way of doing this (like a callback attached to `RenderingMetadata`)